File: public/dsa/searching/non_repeating_element/bits/non_repeating_element.cpp

#include <bits/stdc++.h>
using namespace std;

int findNonRepeatingElement(vector<int> &arr) {
    int xor_value = 0;
    for (int v : arr) {
        xor_value ^= v;
    }
    return xor_value;
}

// tests-start
int main() {
    vector<int> arr = {1, 1, 2, 3, 3, 4, 4};
    cout << findNonRepeatingElement(arr) << endl;
    return 0;
}
// tests-end

/*output
2
*/

File: public/dsa/searching/non_repeating_element/bits/non_repeating_element.py
from typing import List


class Solution:
    def findNonRepeatingElement(self, arr: List[int]) -> int:
        xor = 0
        for v in arr:
            xor ^= v
        return xor


# tests-start
arr = [1, 1, 2, 3, 3, 4, 4]
sol = Solution()
print(sol.findNonRepeatingElement(arr))
# tests-end

'''output
2
'''
